test/dependency_validator/v2_updater_test.dart

```diff
@@ -70,26 +75,32 @@ main() {
               'name: a_package\n'
               'version: 1.3.4\n'
               '\n'
+              '# a comment above environment\n'
               'environment: ">=2.11.0 <3.0.0"\n'
               '\n'
               'dependencies:\n'
+              '  # a comment above a dependency\n'
               '  a_dependency: ^1.0.0\n'
               'dependency_validator:\n'
               '  exclude:\n'
+              '    # a comment above an excluded directory\n'
               '    - app\n'
               '',
           expectedOutput: ''
               'name: a_package\n'
               'version: 1.3.4\n'
               '\n'
+              '# a comment above environment\n'
               'environment: ">=2.11.0 <3.0.0"\n'
               '\n'
               'dependencies:\n'
+              '  # a comment above a dependency\n'
               '  a_dependency: ^1.0.0\n'
-              'dependency_validator: \n'
+              'dependency_validator:\n'
               '  exclude:\n'
+              '    # a comment above an excluded directory\n'
               '    - app\n'
-              '  ignore:\n'
+              '  ignore: \n'
```

Comment:
#nit Why is there a trailing space on this line? Ideally we wouldn't add one

Reply:
That's how the `YamlEditor` adds new keys to the yaml 😞  [This is the condition that does that](https://github.com/Workiva/over_react_codemod/blob/9dcd3539ca23c689305ae2a3da80a80306ca9616/lib/src/dependency_validator_suggestors/v2_updater.dart#L45-L47).